pkg/clients/postgres/clients.go

```diff
@@ -487,47 +475,27 @@ func PageQuery(pm clients.Page) (string, error) {
 	if err != nil {
 		return "", errors.Wrap(errors.ErrMalformedEntity, err)
 	}
-	var query []string
-	var emq string
+
+	query := buildQueryConditions(pm)
 	if mq != "" {
 		query = append(query, mq)
 	}
-	if len(pm.IDs) != 0 {
-		query = append(query, fmt.Sprintf("id IN ('%s')", strings.Join(pm.IDs, "','")))
-	}
-	if pm.Name != "" {
-		query = append(query, "name ILIKE '%' || :name || '%'")
-	}
-	if pm.Identity != "" {
-		query = append(query, "identity ILIKE '%' || :identity || '%'")
-	}
-	if pm.Id != "" {
-		query = append(query, "id ILIKE '%' || :id || '%'")
-	}
-	if pm.Tag != "" {
-		query = append(query, ":tag = ANY(c.tags)")
-	}
+
 	if pm.Status != clients.AllStatus {
 		query = append(query, "c.status = :status")
 	}
 	if pm.Domain != "" {
 		query = append(query, "c.domain_id = :domain_id")
 	}
-
-	if pm.Role != clients.AllRole {
-		query = append(query, "c.role = :role")
-	}
+	var emq string
```

Comment:
At present `RetrieveAll` and `RetrieveByIds` doesn't have ordering . 
I think it is good to have ordering in all list and search function.

Reply:
I can add that